tee-worker/service/src/setup.rs

```diff
@@ -126,6 +137,20 @@ mod needs_enclave {
 	}
 }
 
+/// backs up shard directory and restores it after cleaning shards directory
+pub(crate) fn remove_old_shards(root_dir: &Path, new_shard_name: &str) -> ServiceResult<()> {
+	let shard_backup = root_dir.join("shard_backup");
+	let shard_dir = root_dir.join(SHARDS_PATH).join(new_shard_name);
+
+	fs::rename(shard_dir.clone(), shard_backup.clone()).expect("Failed to backup shard");
+	remove_dir_if_it_exists(root_dir, SHARDS_PATH)?;
+	fs::create_dir_all(root_dir.join(SHARDS_PATH)).expect("Failed to create shards directory");
+	fs::rename(shard_backup, root_dir.join(SHARDS_PATH).join(shard_dir))
```

Comment:
Is this wrong?

`shard_dir` seems to have its base dir already

Reply:
I tested it a bit, it seems `root_dir.join(SHARDS_PATH).join(shard_di)` will remove the duplicated path prefix automatically, but still it might be confusing - so I prefer using shard_dir directly

Reply:
Good catch, it produces the same result that's why the test is ok:

```
println!(">> {:?}", root_dir.join(SHARDS_PATH).join(shard_dir.clone()));
println!(">> {:?}", root_dir.join(SHARDS_PATH).join(new_shard_name));

---- setup::tests::test_remove_old_shards stdout ----
>> "/home/franjs/projects/litentry-parachain/tee-worker/service/test_backup_shard/shards/new_shard"
>> "/home/franjs/projects/litentry-parachain/tee-worker/service/test_backup_shard/shards/new_shard"
```
I'll update it for clarity